Google.PowerShell/Common/ActiveUserCredential.cs

```diff
@@ -0,0 +1,172 @@
+﻿/*
+Copyright 2016 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Auth.OAuth2.Flows;
+using Google.Apis.Http;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.PowerShell.Common
+{
+    /// <summary>
+    /// OAuth 2.0 credential for accessing protected resources using an access token.
+    /// This class will get the access token from "gcloud auth print-access-token".
+    /// </summary>
+    public class ActiveUserCredential : ICredential, IHttpExecuteInterceptor, IHttpUnsuccessfulResponseHandler
```

Comment:
The type name isn't especially useful. `ActiveUserCredential` makes me think it is a data type. However, we  use this class as a substitute for how Apiary authenticates RPC calls. Perhaps we should call this "AuthicateWithSdkCredentialsExecutor"? That name is obviously terrible, but does communicate more of how it is supposed to be used.